### PR TITLE
[MTT-8908] Feat: Adding PigChef character and necessary animations

### PR DESCRIPTION
**Pull request by Elfi0Kuhndorf**
### Description

This PR implements the PigChef character plus its dependencies and the newly created and tweaked animations for the character .
It also has a testscene "CharacterTest" and "ThirdPersonRBController.cs" in order to test the functionality. The controller and scene will be replaced later on by the final Sample scene and controller (see locomotion task).

Make sure to test the characters behaviour to see if all animations (idle, walk, jump. pickup, carry, throw, drop) work as expected.

### Issue Number(s)

https://jira.unity3d.com/browse/MTT-8908

### Contribution checklist
 - [ ] Tests have been added for the project and/or any internal package
 - [ ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=205) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=205) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: Elfi0Kuhndorf<br/>:x: davehunt-unity<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=205) it.</sub>

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs
Comment (fernando-cortez) [inline Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs:1]: For the purposes of testing, and before we integrate the locomotion + animations, this can go through, but let's move this to the Temp/ folder inside of Scripts.

File: Experimental/DistributedAuthoritySample/Assets/Scenes/CharacterTest.unity
Comment (fernando-cortez) [inline Experimental/DistributedAuthoritySample/Assets/Scenes/CharacterTest.unity:1]: Analogously to the testing locomotion script, let's move this to a Temp (or Testing? Test is a reserved path for other purposes) subdirectory inside of Scenes folder.

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs
Comment (fernando-cortez) [inline Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs:125]: @NoelStephensUnity 

I'd get your eyes on this, since this seems to be the inverse of the approach that NetworkRigidbody takes, no?

In this approach, the Avatar character has a Rigidbody at the root, and a child bone in the rig with a Rigidbody & FixedJoint component.

It's worthwhile comparing it to the approach [here](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/feat/distributed-authority-sample/pickup/Experimental/DistributedAuthoritySample/Assets/Scripts/Player/PickupAction.cs), done through NetworkRigidbody.

Could you instead have a NetworkRigidbody on a child GameObject, and attach the item pickup object to that? I'll experiment with that!

**Review by fernando-cortez — COMMENTED** [review]
I'll attach the warnings + a few errors I saw when opening the project after clearing my local library. I'd address the errors and I'll approve once those are cleared up.

Sub Graph at Assets/Shaders/SubGraphs/ToonLightingModel.shadersubgraph has 5 warning(s), the first is: There is a newer version of this node available. Inspect node for details.

Sub Graph at Assets/Shaders/SubGraphs/AdditionalLights.shadersubgraph has 2 warning(s), the first is: There is a newer version of this node available. Inspect node for details.

Sub Graph at Assets/Shaders/SubGraphs/AdditionalLightsToon.shadersubgraph has 1 warning(s), the first is: There is a newer version of this node available. Inspect node for details.

Warning in Graph at Assets/Shaders/Glass.shadergraph on line -1: Ignoring duplicate keyword line 'multi_compile _ LIGHTMAP_ON'

Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (4345119286458520408) doesn't exist!

Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (1597357973998890636) doesn't exist!

Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (1597357973998890636) doesn't exist!

Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (4345119286458520408) doesn't exist!

Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (4345119286458520408) doesn't exist!

File: Experimental/DistributedAuthoritySample/ProjectSettings/TimeManager.asset
Comment (Elfi0Kuhndorf) [reply Experimental/DistributedAuthoritySample/ProjectSettings/TimeManager.asset:6]: This was to test out animations, the timestep can be lowered to see movements in play mode in slow motion. It should be set back to normal now, but I will check and revert the change to develops branch state.

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs
Comment (Elfi0Kuhndorf) [reply Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs:1]: How can I move the script to Temp without it being ignored by github?

File: Experimental/DistributedAuthoritySample/ProjectSettings/TimeManager.asset
Comment (Elfi0Kuhndorf) [reply Experimental/DistributedAuthoritySample/ProjectSettings/TimeManager.asset:6]: Hm I reset them to normal, but it still shows it as a change. @davehunt-unity do these settings look normal to you? Is there something else I should do instead of resetting stats in the "Time" window?

File: Experimental/DistributedAuthoritySample/Assets/Audio.meta
Comment (Elfi0Kuhndorf) [inline Experimental/DistributedAuthoritySample/Assets/Audio.meta:1]: @fernando-cortez with the Audio assets I am not sure if we should delete it all for now or keep it (could implement footsteps and so on at the end). What do you think?

File: Experimental/DistributedAuthoritySample/ProjectSettings/ProjectSettings.asset
Comment (Elfi0Kuhndorf) [reply Experimental/DistributedAuthoritySample/ProjectSettings/ProjectSettings.asset:176]: What I changed here is that I set the Active Input Handling in Player settings to "Both", so that both your player movement scripts and Dave and my temporary one works while we still have it. Later on we can reset it to only the newer Input system.

Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/ProjectSettings/ProjectSettings.asset:176]: It seems like there's a difference in our line endings settings, so the whole file was marked as changed.
Let's resolve that item offline, but if you can add just that one line change instead that'd be great.

File: Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs
Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/Assets/Scripts/Player/ThirdPersonRBController.cs:1]: In Rider, if you navigate to the file itself and stage it, you'll get a prompt to confirm adding an ignored file by Git.
Through the command line you can do `git add -f <path>` to force stage the file.

File: Experimental/DistributedAuthoritySample/Assets/Audio.meta
Comment (fernando-cortez) [reply Experimental/DistributedAuthoritySample/Assets/Audio.meta:1]: Good idea. Less items to review for now.
Let's leave audio to another PR for later.

**Review by richard-harrington — APPROVED** [review]

**Comment by Elfi0Kuhndorf** [comment]
> Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (4345119286458520408) doesn't exist!
> 
> Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (1597357973998890636) doesn't exist!
> 
> Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (1597357973998890636) doesn't exist!
> 
> Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (4345119286458520408) doesn't exist!
> 
> Broken text PPtr in file(Assets/Art/Characters/PigChef/Animation/PigChef.controller). Local file identifier (4345119286458520408) doesn't exist!

Those should be fixed now :D

**Review by fernando-cortez — DISMISSED** [review]
I'm seeing one warning pop up on first project launch related to the character animations:

Main Object Name 'PigChef_ThrowAnticipation_Rig' does not match filename 'PIgChef_ThrowAnticipation_Rig'

Pre-approving since this one seems to be a quick renaming fix.

**Review by fernando-cortez — APPROVED** [review]
Pre-approving as it needs some merges resolved, but the animation warning is gone. LGTM!

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**
